lib/Doctrine/ORM/Tools/SchemaTool.php

```diff
@@ -910,4 +911,42 @@ public function getUpdateSchemaSql(array $classes, $saveMode = false)
 
         return $schemaDiff->toSql($this->platform);
     }
+
+    /**
+     * Creates the schema from the database, ensuring tables from the target schema are whitelisted for comparison.
+     */
+    private function createSchemaForComparison(Schema $toSchema) : Schema
+    {
+        // check and memorize if we are using dbal >= 3.0
+        static $hasFilterExpressionMethod;
+        $hasFilterExpressionMethod = $hasFilterExpressionMethod ?? method_exists(Configuration::class, 'getFilterSchemaAssetsExpression');
+
+        // backup schema assets filter and filter-expression
+        $sm               = $this->em->getConnection()->getSchemaManager();
+        $config           = $this->em->getConnection()->getConfiguration();
+        $filterExpression = $hasFilterExpressionMethod ? $config->getFilterSchemaAssetsExpression() : null;
+        $filter           = $config->getSchemaAssetsFilter();
+
+        if ($filter === null) {
+            return $sm->createSchema();
+        }
+
+        // whitelist assets we already know about in $toSchema, use the existing filter otherwise
+        $config->setSchemaAssetsFilter(static function ($asset) use ($filter, $toSchema) : bool {
+            $assetName = $asset instanceof AbstractAsset ? $asset->getName() : $asset;
+
+            return $toSchema->hasTable($assetName) || $toSchema->hasSequence($assetName) || $filter($asset);
+        });
+
+        try {
+            return $sm->createSchema();
+        } finally {
+            // restore schema assets filter and filter-expression
+            if ($filterExpression !== null) {
```

Comment:
This can be cleaned up if we agree that filterExpression is actually not needed here.